src/smc-hub/hub.coffee

```diff
@@ -1406,7 +1406,7 @@ class Client extends EventEmitter
                         if err
                             cb(err)
                         else if not is_public
-                            cb("project with id '#{mesg.project_id}' is not public")
+                            cb("not_public")
```

Comment:
Since this exact error message is being matched elsewhere, it is CRITICAL to put a comment here saying so and also where it is used.  Otherwise, somebody else could just change the error willy nilly (like you just changed it from what it was).